make_european.py

```diff
@@ -0,0 +1,82 @@
+"""Take input in the format:
+mm/dd/yyyy:username@host
+Print out all of the usernames that have the host 'aol' and convert
+the date to European format so that the following format is produced:
+dd/mm/yyyy username
+"""
+import re
+
+def get_input():
+    """Prompts the user for input in the format:
+    mm/dd/yyyy:username@host
+    Receive the input as raw_input.
+    """
+    try:
+        return raw_input()
+    except EOFError:
+        return ' '
+
+def convert_input(string, desired_host):
+    """Take in a string and a desired host. If the host found in the
+    string matches the desired host, then convert the found date to the
+    European format and convert the username to all lowercase. Return
+    the new date and username.
+    """
+    assert isinstance(string, str), "Initial input for convert_input needs to be a string."
+    assert isinstance(desired_host, str), "Host variable for convert_input needs to be a string."
+
+    username, host = find_email_address(string)
+    if host.lower() == desired_host:
+        month, day, year = find_date(string)
+        euro_date = '%s/%s/%s' % (day, month, year) # Convert the date into European format.
+        return '%s %s' % (euro_date, username.lower())
+
+def find_email_address(string):
+    """Take in a string and return the username and host of the email
+    address found in the string.
+    """
+    assert isinstance(string, str), "The input for find_email_address needs to be a string."
+
+    email_address = re.search(r'([\w.-]+)@([\w.-]+)', string)
+    assert email_address, "A proper email address must be provided."
```

Comment:
this is a great assertion.

also, check out: http://www.regular-expressions.info/email.html

i'm going to judge your own code by it passing the tests - its a great example of YAGNI. none of the emails i provided beat your regex, so that regex is good enough for our requirements. BUT, matching an email is notoriously hard, so go ahead and read that for some surrounding thoughts.